From 7c512b1ec835c3a2dea9290846e30f1877ff913f Mon Sep 17 00:00:00 2001
From: "Donald.E.Lippi" <donald.e.lippi@noaa.gov>
Date: Tue, 21 Apr 2026 18:40:21 +0000
Subject: [PATCH] Workaound: update surface fields in mpas_state_mod

---
 src/mpasjedi/State/mpas_state_mod.F90 | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/src/mpasjedi/State/mpas_state_mod.F90 b/src/mpasjedi/State/mpas_state_mod.F90
index 7e3cdb4..350e3b8 100644
--- a/src/mpasjedi/State/mpas_state_mod.F90
+++ b/src/mpasjedi/State/mpas_state_mod.F90
@@ -72,11 +72,13 @@ subroutine add_incr(self, increment)
    integer :: i, k, ngrid
    type (mpas_pool_type), pointer :: state, diag, mesh
    type (field2DReal), pointer :: fld2d_pb, fld2d_u, fld2d_u_inc, fld2d_uRm, fld2d_uRz
+   type (field1DReal), pointer :: fld1d_u10 => null(), fld1d_v10 => null()
    type (field2DReal), pointer :: fld2d_p, fld2d_dp, fld2d_drho, fld2d_dth, fld2d_dqv
    real(kind=RKIND), dimension(:,:), pointer :: ptrr2_qv, ptrr2_sh
    real(kind=RKIND), dimension(:,:), pointer :: ptrr2_p, ptrr2_rho, ptrr2_t, ptrr2_th, ptrr2_pp
    real(kind=RKIND), dimension(:,:), pointer :: ptrr2_dp, ptrr2_drho, ptrr2_dt, ptrr2_dth, ptrr2_dsh
    real(kind=RKIND), dimension(:), pointer :: ptrr1_ps, ptrr1_dps
+   real(kind=RKIND), dimension(:), pointer :: ptrr1_2mt => null(), ptrr1_2msh => null()
 
    ! Difference with self_add other is that self%subFields can contain extra fields
    ! beyond increment%subFields and the resolution of increment can be different.
@@ -155,6 +157,17 @@ subroutine add_incr(self, increment)
       kind_op = 'add'
       call da_operator(trim(kind_op), self%subFields, increment%subFields, fld_select = increment%fldnames_ci)
 
+      ! Third, update thermodynamic subFields at surface
+      if (self%has('air_temperature_at_2m') .and. associated(ptrr2_dt)) then
+         call self%get('air_temperature_at_2m', ptrr1_2mt)
+         ptrr1_2mt(1:ngrid)  = ptrr1_2mt(1:ngrid) + ptrr2_dt(1,1:ngrid)
+      endif
+      if (self%has('water_vapor_mixing_ratio_wrt_moist_air_at_2m') .and. &
+          associated(ptrr2_dsh)) then
+         call self%get('water_vapor_mixing_ratio_wrt_moist_air_at_2m', ptrr1_2msh)
+         ptrr1_2msh(1:ngrid) = ptrr1_2msh(1:ngrid) + ptrr2_dsh(1,1:ngrid)
+      endif
+
       ! Impose positive-definite limits on hydrometeors and moistureFields
       ! note: nonlinear change of variable
       call da_posdef( self%subFields, mpas_hydrometeor_fields)
@@ -205,6 +218,15 @@ subroutine add_incr(self, increment)
          ! TODO: DO we need HALO exchange here or in ModelMPAS::initialize for model integration?
          call mpas_deallocate_field( fld2d_u_inc )
       endif
+      !Update dynamic subFields at surface
+      !if (self%has('eastward_wind_at_10m') .and. &
+      !    self%has('northward_wind_at_10m') .and. &
+      !    associated(fld2d_uRz) .and. associated(fld2d_uRm)) then
+      !   call mpas_pool_get_field(self%subFields, 'eastward_wind_at_10m', fld1d_u10)
+      !   call mpas_pool_get_field(self%subFields, 'northward_wind_at_10m', fld1d_v10)
+      !   fld1d_u10%array(1:ngrid) = fld1d_u10%array(1:ngrid) + fld2d_uRz%array(1,1:ngrid)
+      !   fld1d_v10%array(1:ngrid) = fld1d_v10%array(1:ngrid) + fld2d_uRm%array(1,1:ngrid)
+      !endif
    else
       call abor1_ftn("mpas_state:add_incr: dimension mismatch")
    endif
-- 
2.43.7

