From 96519962ce0d0b102d0f177972efb605ccbecd74 Mon Sep 17 00:00:00 2001
From: Samuel Degelia <samuel.degelia@dlogin05.dogwood.wcoss2.ncep.noaa.gov>
Date: Wed, 27 May 2026 16:22:15 +0000
Subject: [PATCH] Workaround: update 10 m wind fields in mpas_state_mod

---
 src/mpasjedi/State/mpas_state_mod.F90 | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/mpasjedi/State/mpas_state_mod.F90 b/src/mpasjedi/State/mpas_state_mod.F90
index e3dffc2..a459f08 100644
--- a/src/mpasjedi/State/mpas_state_mod.F90
+++ b/src/mpasjedi/State/mpas_state_mod.F90
@@ -71,6 +71,7 @@ subroutine add_incr(self, increment)
    integer :: i, k, ngrid
    type (mpas_pool_type), pointer :: state, diag, mesh
    type (field2DReal), pointer :: fld2d_pb, fld2d_u, fld2d_u_inc, fld2d_uRm, fld2d_uRz
+   type (field1DReal), pointer :: fld1d_u10 => null(), fld1d_v10 => null()
    type (field2DReal), pointer :: fld2d_p, fld2d_dp, fld2d_drho, fld2d_dth, fld2d_dqv
    real(kind=RKIND), dimension(:,:), pointer :: ptrr2_qv, ptrr2_sh
    real(kind=RKIND), dimension(:,:), pointer :: ptrr2_p, ptrr2_rho, ptrr2_t, ptrr2_th, ptrr2_pp
@@ -213,6 +214,15 @@ subroutine add_incr(self, increment)
          ! TODO: DO we need HALO exchange here or in ModelMPAS::initialize for model integration?
          call mpas_deallocate_field( fld2d_u_inc )
       endif
+      !Update dynamic subFields at surface
+      !if (self%has('eastward_wind_at_10m') .and. &
+      !    self%has('northward_wind_at_10m') .and. &
+      !    associated(fld2d_uRz) .and. associated(fld2d_uRm)) then
+      !   call mpas_pool_get_field(self%subFields, 'eastward_wind_at_10m', fld1d_u10)
+      !   call mpas_pool_get_field(self%subFields, 'northward_wind_at_10m', fld1d_v10)
+      !   fld1d_u10%array(1:ngrid) = fld1d_u10%array(1:ngrid) + fld2d_uRz%array(1,1:ngrid)
+      !   fld1d_v10%array(1:ngrid) = fld1d_v10%array(1:ngrid) + fld2d_uRm%array(1,1:ngrid)
+      !endif
    else
       call abor1_ftn("mpas_state:add_incr: dimension mismatch")
    endif
-- 
2.43.7

