From bbd66f7f07b18099dcabff85ddde3a2f669cf430 Mon Sep 17 00:00:00 2001
From: "Donald.E.Lippi" <donald.e.lippi@noaa.gov>
Date: Fri, 10 Apr 2026 01:34:06 +0000
Subject: [PATCH 3/3] Workaround: add q2m metadata (fv3jedi)

---
 src/fv3jedi/FieldMetadata/FieldsMetadataDefault.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/fv3jedi/FieldMetadata/FieldsMetadataDefault.h b/src/fv3jedi/FieldMetadata/FieldsMetadataDefault.h
index 10e42059..acc9bfde 100644
--- a/src/fv3jedi/FieldMetadata/FieldsMetadataDefault.h
+++ b/src/fv3jedi/FieldMetadata/FieldsMetadataDefault.h
@@ -187,6 +187,14 @@ namespace fv3jedi {
     md.space = "magnitude";
     addFieldMetadata(fieldsmetadata, nlev, md);
 
+    md.longName = "water_vapor_mixing_ratio_wrt_moist_air_at_2m";
+    md.units = "kgkg-1";
+    md.kind = "double";
+    md.tracer = "true";
+    md.levels = "1";
+    md.space = "magnitude";
+    addFieldMetadata(fieldsmetadata, nlev, md);
+
     md.longName = "saturation_water_vapor_mixing_ratio_wrt_moist_air";
     md.units = "kgkg-1";
     md.kind = "double";
-- 
2.43.7

